aio/aio-proxy/aio_proxy/search/execute_search.py

```diff
@@ -62,5 +48,71 @@ def sort_and_execute_search(
             matching_unite_legale_dict["matching_etablissements"] = []
 
         response.append(matching_unite_legale_dict)
+    search_response = {
+        "total_results": total_results,
+        "response": response,
+        "execution_time": execution_time,
+        "include_etablissements": include_etablissements,
+    }
+    return search_response
+
+
+def sort_search(search, is_text_search: bool):
+    # Sorting is very heavy on performance if there is no
+    # search terms (only filters). As there is no search terms, we can
+    # exclude this sorting because score is the same for all results
+    # documents. Beware, nom and prenoms are search fields.
+    if is_text_search:
+        search = search.sort(
+            {"_score": {"order": "desc"}},
+            {"etat_administratif_unite_legale": {"order": "asc"}},
+        )
+    # If only filters are used, use nombre établissements ouverts to sort the results
+    else:
+        search = search.sort(
+            {"nombre_etablissements_ouverts": {"order": "desc"}},
+        )
+    return search
+
+
+def sort_and_execute_search(
+    search,
+    offset: int,
+    page_size: int,
+    is_text_search: bool,
+    include_etablissements: bool,
+) -> dict:
+    search = search.extra(track_scores=True)
+    search = search.extra(explain=True)
+    # Collapse is used to aggregate the results by siren. It is the consequence of
+    # separating large documents into smaller ones
+    search = search.update_from_dict({"collapse": {"field": "siren"}})
+    search = sort_search(search, is_text_search)
+
+    # Execute search
+    def get_search_response():
+        return execute_and_format_search_response(
+            search,
+            offset,
+            page_size,
+            include_etablissements,
+        )
+
+    search_response = cache_strategy(
+        search,
+        get_search_response,
+        should_cache_search_response,
```

Comment:
why do you pass that function in parameter ? Can you not put it in `cache.py`. It seems better there for me.

Reply:
`should_cache_search_response` is related to search execution via the `execution_time` parameter. And it is in theory independent from the cache running strategy.